src/m3ninx/index/segment/builder/multi_segments_builder.go

```diff
@@ -147,6 +150,11 @@ func (b *builderFromSegments) AddSegments(segments []segment.Segment) error {
 			d := iter.Current()
 			negativeOffsets = append(negativeOffsets, currOffset)
 			if b.idSet.Contains(d.ID) {
+				if d.OnIndexSeries != nil {
+					// Reconcile any duplicate series.
```

Comment:
nit: maybe more detailed comment here on why it is important to reconcile these (i.e. primarily merge the time ranges)